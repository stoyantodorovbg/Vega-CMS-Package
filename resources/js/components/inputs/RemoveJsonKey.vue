<template>
    <button v-if="this.input_key !== 'empty_json'"
        type="button"
        class="btn btn-danger btn-sm"
        @click="removeKey()"
    >Remove {{ title }} input key</button>
</template>

<script>

    export default {
        name: 'RemoveJsonKey',

        props: {
            title: {
                type: String,
                default: ''
            },
            input_key: {
                type: String,
                default: ''
            }
        },

        methods: {
            removeKey() {
                if (this.input_key !== '') {
                    this.$parent.removeStructureKey(this.input_key);
                }
            }
        }
    }
</script>
