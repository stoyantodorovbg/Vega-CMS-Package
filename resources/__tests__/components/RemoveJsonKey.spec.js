import { mount } from '@vue/test-utils'
import RemoveJsonKey from "../../js/components/inputs/RemoveJsonKey";

describe('RemoveJsonKey Component', () => {
  test('should mount without crashing', () => {
    const wrapper = mount(RemoveJsonKey);

    expect(wrapper).toMatchSnapshot();
  });
});
