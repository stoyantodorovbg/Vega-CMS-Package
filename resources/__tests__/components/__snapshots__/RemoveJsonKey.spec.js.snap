// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RemoveJsonKey Component should mount without crashing 1`] = `
VueWrapper {
  "_emitted": Object {},
  "_emittedByOrder": Array [],
  "isFunctionalComponent": undefined,
}
`;
